Tools/Scripts/libraries/webkitapipy/webkitapipy/macho.py

```diff
@@ -0,0 +1,171 @@
+                in_section = Sect.SELREFS
+            elif line == '(__TEXT,__dlsym_cstr) section:':
+                in_section = Sect.DLSYM
+            elif line == '(__TEXT,__getClass_cstr) section:':
+                in_section = Sect.GETCLASS
+
+            # Parse symbol information based on the current section.
+            elif in_section == Sect.EXPORTS:
+                # Address in binary and symbol name:
+                # ```
+                # 0x00A5B75C  _JSBigIntCreateWithDouble
+                # ```
+                offset, symbol = line.split(maxsplit=1)
+                # skip the header line
+                if offset != 'offset':
+                    self.exports.add(symbol)
+            elif in_section == Sect.IMPORTS:
+                # Hexadecimal index, symbol name, optional linkage tag, and
+                # containing dylib as recorded in the two-level namespace:
+                # ```
+                # 0x02E0  _voucher_mach_msg_set [weak-import] (from libSystem)
+                # ```
+                idx, symbol, metadata = line.split(maxsplit=2)
+                dylib = metadata[metadata.index('(from ') + 6:-1]
+                if dylib != '<this-image>':
+                    self.imports.add(symbol)
+            elif in_section == Sect.OBJC:
+                # ObjC-like declaration for classes and protocols with method
+                # names and method addresses for classes:
+                # ```
+                # @interface WebEventRegion : NSObject <NSCopying>
+                #   0x02134AC4  -[WebEventRegion initWithPoints::::]
+                #   0x02134B80  -[WebEventRegion copyWithZone:]
+                #   0x02134BB8  -[WebEventRegion description]
+                #   0x02134C58  -[WebEventRegion hitTest:]
+                #   0x02134CD4  -[WebEventRegion isEqual:]
+                #   0x02134E94  -[WebEventRegion quad]
+                #   0x02134F74  -[WebEventRegion p1]
+                #   0x02134F9C  -[WebEventRegion p2]
+                #   0x02134FC4  -[WebEventRegion p3]
+                #   0x02134FEC  -[WebEventRegion p4]
+                # @end
+                # @protocol NSCopying :
+                #   -[NSCopying copyWithZone:]
+                # @end
+                # ```
+                m = objc_fully_qualified_method.search(line)
+                if m:
+                    self.methods.add(m.group('selector'))
+                elif '@interface' not in line and \
+                        '@protocol' not in line and '@end' not in line:
+                    print(f'warning:{self.file} unrecognized '
+                          f'dyld_info -objc line: "{line}"', file=sys.stderr)
+            elif in_section == Sect.SELREFS:
+                # Address in binary of selref data, followed by selector
+                # name:
+                # ```
+                # 0x0A8043D8  "interruption:"
+                # ```
+                address, quoted_name = line.split(maxsplit=1)
```

Comment:
Given an unexpected line without any whitespace, this would throw. Maybe that's fine?

Reply:
I think it's fine, at least without evidence that this is a real risk. I'd rather it throw an exception than semi-silently discard binary information, since that could lead to our failing to detect SPI.